server/src/routes/projectsRoute.ts

```diff
@@ -5,15 +5,15 @@ import {
   editProject,
   getSpecificProject,
 } from '../controllers';
-import { checkAuth } from '../middlewares';
+import { asyncMiddleware, checkAuth } from '../middlewares';
 
 const projectsRoute = require('express').Router();
 
 projectsRoute.use(checkAuth);
```

Comment:
check auth also has try and catch, so we can wrap our middleware with async-middleware and remove try-catch block